beacon-chain/rpc/beaconv1/blocks_test.go

```diff
@@ -0,0 +1,399 @@
+	genesisRoot, err := genesis.Block.HashTreeRoot()
+	require.NoError(t, err)
+	require.NoError(t, db.SaveState(ctx, beaconState, genesisRoot), "Could not save genesis state")
+
+	c := &mock.ChainService{Root: bsRoot[:], State: beaconState}
+	beaconChainServer := &Server{
+		BeaconDB:          db,
+		ChainStartFetcher: &mockPOW.POWChain{},
+		BlockReceiver:     c,
+		HeadFetcher:       c,
+		BlockNotifier:     c.BlockNotifier(),
+		Broadcaster:       mockp2p.NewTestP2P(t),
+	}
+	req := testutil.NewBeaconBlock()
+	req.Block.Slot = 5
+	req.Block.ParentRoot = bsRoot[:]
+	v1Block, err := v1alpha1ToV1Block(req)
+	require.NoError(t, err)
+	require.NoError(t, db.SaveBlock(ctx, req))
+	blockReq := &ethpb.BeaconBlockContainer{
+		Message:   v1Block.Block,
+		Signature: v1Block.Signature,
+	}
+	_, err = beaconChainServer.SubmitBlock(context.Background(), blockReq)
+	assert.NoError(t, err, "Could not propose block correctly")
+}
+
+func TestServer_GetBlock_GenesisRoot(t *testing.T) {
+	db, _ := dbTest.SetupDB(t)
+	ctx := context.Background()
+
+	bs := &Server{
+		BeaconDB: db,
+	}
+
+	genBlk, _ := fillDBTestBlocks(t, ctx, db)
+	root, err := genBlk.Block.HashTreeRoot()
+	require.NoError(t, err)
+
+	// Should throw an error if more than one blk returned.
+	block, err := bs.GetBlock(ctx, &ethpb.BlockRequest{
+		BlockId: root[:],
+	})
+	require.NoError(t, err)
+
+	marshaledBlk, err := genBlk.Block.Marshal()
+	require.NoError(t, err)
+	v1Block := &ethpb.BeaconBlock{}
+	require.NoError(t, proto.Unmarshal(marshaledBlk, v1Block))
+
+	if !reflect.DeepEqual(block.Data.Message, v1Block) {
+		t.Error("Expected blocks to equal")
+	}
+}
+
+func TestServer_GetBlock_Genesis(t *testing.T) {
+	db, _ := dbTest.SetupDB(t)
+	ctx := context.Background()
+
+	bs := &Server{
+		BeaconDB: db,
+	}
+
+	genBlk, _ := fillDBTestBlocks(t, ctx, db)
+
+	// Should throw an error if more than one blk returned.
+	block, err := bs.GetBlock(ctx, &ethpb.BlockRequest{
+		BlockId: []byte("genesis"),
+	})
+	require.NoError(t, err)
+
+	marshaledBlk, err := genBlk.Block.Marshal()
+	require.NoError(t, err)
+	v1Block := &ethpb.BeaconBlock{}
+	require.NoError(t, proto.Unmarshal(marshaledBlk, v1Block))
+
+	if !reflect.DeepEqual(block.Data.Message, v1Block) {
+		t.Error("Expected blocks to equal")
+	}
+}
+
+func TestServer_GetBlock_Head(t *testing.T) {
+	db, _ := dbTest.SetupDB(t)
+	ctx := context.Background()
+
+	bs := &Server{
+		BeaconDB: db,
+	}
+
+	_, blkContainers := fillDBTestBlocks(t, ctx, db)
+
+	headBlock := blkContainers[len(blkContainers)-1]
+	bs = &Server{
+		BeaconDB:    db,
+		HeadFetcher: &mock.ChainService{DB: db, Block: headBlock.Block},
+	}
+	// Should throw an error if more than one blk returned.
+	block, err := bs.GetBlock(ctx, &ethpb.BlockRequest{
+		BlockId: []byte("head"),
+	})
+	require.NoError(t, err)
+
+	marshaledBlk, err := headBlock.Block.Block.Marshal()
+	require.NoError(t, err)
+	v1Block := &ethpb.BeaconBlock{}
+	require.NoError(t, proto.Unmarshal(marshaledBlk, v1Block))
```

Comment:
Is this v1ToV1alpha1Block? I see these 4 lines of code repeated several times. May be fixed by table drive tests and reuse of code.